### ISSUE TITLE
Issue with AuthMe and proxy-support

### ISSUE THREAD

**Issue by yangm97** [GlowstoneMC/Glowstone#547]
Looking at the logs, both Bungeecord and Glowstone are receiving the correct player IP:

```
06:28:30 [INFO] [/187.180.212.96:39420] <-> InitialHandler has connected
06:28:30 [INFO] [TheticalFlyer97] <-> ServerConnector [lobby] has connected
```

```
06:28:30 [INFO] UUID of player TheticalFlyer97 is a6462498-08d5-37c4-bfbb-158cac0008f2
06:28:30 [INFO] TheticalFlyer97 [/187.180.212.96:43947] connected, UUID: a6462498-08d5-37c4-bfbb-158cac0008f2
06:28:30 [INFO] TheticalFlyer97 joined the game
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: FML
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: AuthMeBridge
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: FML|HS
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: FORGE
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: BTLP/Bridge
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: BungeeCord
06:28:30 [INFO] TheticalFlyer97[/187.180.212.96:43947] registered channel: BAuthMeBridge
06:28:42 [INFO] TheticalFlyer97 issued command: /register (censored)
```

But looking at the AuthMe database, you can see that AuthMe has, somehow, received the Bungeecord IP from Glowstone and is using that instead of the player IP:

![screen shot 2017-08-13 at 03 55 16](https://user-images.githubusercontent.com/4412002/29246869-60ee637c-7fdb-11e7-9d87-807093f5de8e.png)

Another issue with AuthMe is that it’s not censoring commands with passwords (like /register and /login). AuthMe works perfectly fine otherwise.

**Comment by mastercoms** [comment]
It won't be able to censor commands unless it adds support for Glowstone. It uses a different logging interface than what Glowstone uses. The IP problem is strange, though. Also, you may want to censor your IP :).

**Comment by yangm97** [comment]
My IP is dynamic, so it has probably changed itself a number of times already but thanks :)

I will give a shout to AuthMe devs regarding the logging interface then.

**Comment by Postremus** [comment]
I was not able to reproduce the problem that authme uses the bungeecord ip.
Have you set `proxy-support` to true?

**Comment by yangm97** [comment]
Yes. As you can see, the forwarded IP can be seen on GlowStone logs.

I’m afraid this might be somehow related to the way networking works in docker swarm. I’ve seen some issues on their side too, but as soon as I changed the port mapping mode to “host”, both bungee and glowstone started seeing the player IP (see  [this comment](https://github.com/moby/moby/issues/25526#issuecomment-274810667)).

**Comment by mastercoms** [comment]
Relevant AuthMe code: https://github.com/AuthMe/AuthMeReloaded/blob/1f8307c8f6457741a130bccecc1d75977d99fe54/src/main/java/fr/xephi/authme/util/PlayerUtils.java#L39

**Comment by yangm97** [comment]
[temp.zip](https://github.com/GlowstoneMC/Glowstone/files/1236587/temp.zip)

This is my (stripped down) server. Removed most plugins and config so you can deploy only the relevant parts (lobby(glowstone), bungeecord and database).
Build with: `docker-compose build` and deploy with `docker stack deploy -c docker-compose.yml whateverstackname`. Bungeecord will be listening on the default minecraft port. I recommend testing with an external client, but I think you can replicate the issue from localhost too.

**Comment by VaiTon** [comment]
@yangm97 are you able to reproduce with the latest build?

**Comment by ahdg6** [comment]
In latest build,I can't even use Authme on glowstone.
Thu Jul 01 11:58:29 UTC 2021:https://ci.codemc.io/job/AuthMe/job/AuthMeReloaded/
https://paste.ubuntu.com/p/dFYFzTNdQb/

**Comment by yangm97** [comment]
@VaiTon good question. Has anyone tried to use the reproduction docker stack with the latest build?

**Comment by mastercoms** [comment]
> In latest build,I can't even use Authme on glowstone.
> Thu Jul 01 11:58:29 UTC 2021:https://ci.codemc.io/job/AuthMe/job/AuthMeReloaded/
> https://paste.ubuntu.com/p/dFYFzTNdQb/

That's related to this: #914

**Comment by mastercoms** [comment]
I haven't used the Docker container yet, but I can test this out soon.
